src/CommonITILSatisfaction.php

```diff
@@ -331,4 +353,76 @@ public static function getFormURLWithID($id = 0, $full = true)
         $itemtype = static::getItemtype();
         return $itemtype::getFormURLWithID($satisfaction->fields[$itemtype::getForeignKeyField()]) . '&forcetab=' . $itemtype::getType() . '$3';
     }
+
+    public static function rawSearchOptionsToAdd(string $table, $base_id = 0)
```

Comment:
I do not like the idea of changing base signature for this. It will force developpers to pass a new param to get expected SO ids. Maybe using an abstract getSearchOptionStartingIndex could be a better solution.

Reply:
Ok, I'll look into adding `getSearchOptionStartingIndex`.
Is there any plan to move away from these number based search options id in the future ?

Reply:
> Is there any plan to move away from these number based search options id in the future ?

We could use some kind of nameid (e.g. `id`, `name`, `satisfaction-rate`, ...), but we would have to find relevant name for every search option, and code update would be a real pain. I am not against such a change, but I am not sure anyone will be vonlunteer for this if we decide to do it.

Reply:
Maybe using something like this (not tested):
```diff
diff --git a/src/Search.php b/src/Search.php
index bd3ff9c92e..83df8fe645 100644
--- a/src/Search.php
+++ b/src/Search.php
@@ -3933,16 +3933,22 @@ JAVASCRIPT;
             && count($searchopt[$ID]["additionalfields"])
         ) {
             foreach ($searchopt[$ID]["additionalfields"] as $key) {
+                if (preg_match('/^TABLE\./', $key) === 1) {
+                    $key = preg_replace('/^TABLE\./', '', $key);
+                    $additionalfield_field = $DB->quoteName($orig_table . '.' . $key);
+                } else {
+                    $additionalfield_field = $DB->quoteName($table . $addtable . '.' . $key);
+                }
                 if (
                     $meta
                     || (isset($searchopt[$ID]["forcegroupby"]) && $searchopt[$ID]["forcegroupby"])
                 ) {
-                    $ADDITONALFIELDS .= " IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`$table$addtable`.`$key`,
+                    $ADDITONALFIELDS .= " IFNULL(GROUP_CONCAT(DISTINCT CONCAT(IFNULL($additionalfield_field,
                                                                          '" . self::NULLVALUE . "'),
                                                    '" . self::SHORTSEP . "', $tocomputeid)ORDER BY $tocomputeid SEPARATOR '" . self::LONGSEP . "'), '" . self::NULLVALUE . self::SHORTSEP . "')
                                     AS `" . $NAME . "_$key`, ";
                 } else {
-                    $ADDITONALFIELDS .= "`$table$addtable`.`$key` AS `" . $NAME . "_$key`, ";
+                    $ADDITONALFIELDS .= "$additionalfield_field AS `" . $NAME . "_$key`, ";
                 }
             }
```